### PR TITLE
adds input value as fn

### PR DESCRIPTION
**Pull request by AutoSponge**
see #95

### CODE REVIEW

**Comment by busypeoples** [comment]
Thanks @AutoSponge!
Will run this tomorrow and then merge.

File: src/index.js [hunk 1]
```diff
@@ -42,15 +42,18 @@ const runPredicate = ([predicate, errorMsg]:[Function, string],
...
  * @param {Object} spec the rule object
- * @param {Object} input the validation input data
+ * @param {Object|Function} input the validation input data
  * @returns {{}}
  */
-export const validate = curry((successFn: Function, failFn: Function, spec: Object, input: Object): Object =>
-  reduce((result, key) => {
-    const value = input[key]
+export const validate = curry((successFn: Function, failFn: Function, spec: Object, input: Object): Object => {
+  const inputFn = typeof input === 'function' ? input : (key?: string) => key ? input : input
```
Comment (busypeoples) [inline src/index.js]: `input : (key?: string) => key ? input : input`

**Comment by busypeoples** [comment]
This is good!

File: src/index.js (hunk 1 above)
Comment (AutoSponge) [reply src/index.js]: It's clearer in the test. `key` is optional as you can also provide a thunk or bound fn that ignores `key`.

Comment (busypeoples) [reply src/index.js]: Ok, let's keep the `key ? input : input` part, I don't mind and adapt when needed. Just want to make sure, it's not confusing to people reading the code.

Comment (busypeoples) [reply src/index.js]: Will merge tomorrow and release a new version. Thanks for all the work @AutoSponge!

**Pull request merged**